### DISCUSSION TITLE
AI message to popup window

### DISCUSSION THREAD

**Discussion by gsuuon** [gsuuon/llm.nvim#15] [Q&A]
original: https://github.com/gsuuon/llm.nvim/issues/13#issuecomment-1734297812

@gsuuon is there way to output AI message to popup window? 

https://github.com/gsuuon/llm.nvim/assets/13521338/f923bb23-6374-4cfd-88e7-809613e9b440

The issue I have now - llama will output code with comments, and I cant seem to force it to output the pure code only. 
It  would help, if  output was written to new popup windwo, where user could copy the code, close popup, and paste into place...
I was also wondering maybe: LlmDelete - shoudl be renamed LlmUndo - since this is what it is for

**Comment by gsuuon** [comment]
@JoseConseco moved here to keep the other issue on-topic, feel free to continue asking questions here or create a new discussion.

The markdown parsing would be part of the prompt's transform field - it's built into the [code starter](https://github.com/gsuuon/llm.nvim/blob/a4b4f32d4d4c7424b957fbae7979983eb513ab74/lua/llm/prompts/starters.lua#L123) as an example. It'll automatically remove everything in the response segment that isn't in a markdown block, and everything else goes to `vim.notify`.

About renaming LlmDelete - undo implies a sequence, delete is just removing the only completion there (there's no history, there's just one item). I'm not stuck on the naming though, feel free to open a discussion or issue to get feedback.

**Comment by JoseConseco** [comment]
Thx, it worked ok.  Still it would be cool if we cold redirect output to new popup window - eg. If user ask question, it should not pollute the current code with answer. 
How about LlmRevert  ? Or LlmRestore. In my case if I use 'replace' method, then  LlmDelete is not really deleting last server response, but rather it is reverting code block to original state.

**Reply by gsuuon** [reply]
I guess you're looking for the [`buffer`](https://github.com/gsuuon/llm.nvim/blob/a4b4f32d4d4c7424b957fbae7979983eb513ab74/lua/llm/provider.lua#L26) mode? Check out the [ask](https://github.com/gsuuon/llm.nvim/blob/a4b4f32d4d4c7424b957fbae7979983eb513ab74/lua/llm/prompts/starters.lua#L146) starter example. It copies over input with your question and shows everything in a new buffer.

LlmRestore sounds like it would be confusing in other ways, like it might restore a previous Llm response. Same with LlmRevert. LlmDelete clearly only ever deletes a response.
